src/backend/distributed/operations/worker_node_manager.c

```diff
@@ -499,6 +499,18 @@ DistributedTablePlacementNodeList(LOCKMODE lockMode)
 }
 
 
+bool
+IsSingleNodeCluster(void)
+{
+	if (list_length(DistributedTablePlacementNodeList(NoLock)) < 2)
```

Comment:
If `DistributedTablePlacementNodeList` is zero, it is probably not considered single node right? I guess we could check directly if it is equal to 1 here. But then it will return false if there is only coordinator and `shouldhaveshards` is false. I don't know what that setup is called. As in if the coordinator can have reference tables but not distributed tables, is it a single node cluster or is it not a cluster at this point